test/test_mq.rs

```diff
@@ -79,28 +97,33 @@ fn test_mq_setattr() {
     };
     let mqd = r.unwrap();
 
-    let new_attr =  MqAttr::new(0, 20, MSG_SIZE * 2, 100);
+    let new_attr = MqAttr::new(0, 20, MSG_SIZE * 2, 100);
     let old_attr = mq_setattr(mqd, &new_attr).unwrap();
-    assert_eq!(old_attr, initial_attr);
+    assert_attr_eq!(old_attr, initial_attr);
 
-    let new_attr_get = mq_getattr(mqd).unwrap();
-    // The following tests make sense. No changes here because according to the Linux man page only
+    // No changes here because according to the Linux man page only
     // O_NONBLOCK can be set (see tests below)
-    assert_ne!(new_attr_get, new_attr);
+    #[cfg(not(any(target_os = "dragonfly", target_os = "netbsd")))]
+    {
+        let new_attr_get = mq_getattr(mqd).unwrap();
+        assert_ne!(new_attr_get, new_attr);
```

Comment:
Presumably we could enable this assert for NetBSD and DragonFly as well, but writing the negation of `assert_attr_eq` is just painful enough (and IMO low-value enough) that I'd prefer not to do it. :)